codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/serialize/JsonSerializerGenerator.kt

```diff
@@ -297,30 +297,30 @@ class JsonSerializerGenerator(
         }
     }
 
-    private fun RustWriter.jsonArrayWriter(context: MemberContext, inner: RustWriter.(String) -> Unit) {
+    fun RustWriter.jsonArrayWriter(context: MemberContext, inner: RustWriter.(String) -> Unit) {
         safeName("array").also { arrayName ->
             rust("let mut $arrayName = ${context.writerExpression}.start_array();")
             inner(arrayName)
             rust("$arrayName.finish();")
         }
     }
 
-    private fun RustWriter.jsonObjectWriter(context: MemberContext, inner: RustWriter.(String) -> Unit) {
+    fun RustWriter.jsonObjectWriter(context: MemberContext, inner: RustWriter.(String) -> Unit) {
         safeName("object").also { objectName ->
             rust("let mut $objectName = ${context.writerExpression}.start_object();")
             inner(objectName)
             rust("$objectName.finish();")
         }
     }
 
-    private fun RustWriter.serializeCollection(context: Context<CollectionShape>) {
+    fun RustWriter.serializeCollection(context: Context<CollectionShape>) {
```

Comment:
might be missing something, but I don't see any usages of these functions that would necessitate making them public

Reply:
You are right, there should be private, it is a leftover from some testing I did that I forgot to change back. Will fix.